definitions/checks/foreman_installer/installer_noop_run.rb

```diff
@@ -0,0 +1,22 @@
+module Checks
+  module ForemanInstaller
+    class InstallerNoopRun < ForemanMaintain::Check
+      metadata do
+        description 'Run the installer in noop mode for system checks'
+        tags :pre_upgrade
+      end
+
+      def run
+        check_error_codes = [20, 102, 103]
+        cmd_status, output = feature(:installer).run_with_status(' -n')
```

Comment:
```suggestion
        cmd_status, output = feature(:installer).run_with_status(' --noop')
```

nitpick, but long options are always easier to read ;)